crates/telio-wg/src/link_detection/mod.rs

```diff
@@ -315,48 +256,26 @@ impl StateDecision {
 }
 
 struct State {
-    stats: BytesAndTimestamps,
+    stats: Arc<Mutex<BytesAndTimestamps>>,
     variant: StateVariant,
 }
 
 impl State {
     const POSSIBLE_DOWN_DELAY: Duration = Duration::from_secs(3);
 
-    fn new(rx_bytes: Option<u64>, tx_bytes: Option<u64>, node_state: NodeState) -> Self {
-        let stats = BytesAndTimestamps::new(rx_bytes, tx_bytes);
-        if matches!(node_state, NodeState::Connected) {
-            State {
-                stats,
-                variant: StateVariant::Up,
-            }
-        } else {
-            State {
-                stats,
-                variant: StateVariant::Down,
-            }
+    fn new(stats: Arc<Mutex<BytesAndTimestamps>>, rtt: Duration) -> Self {
+        State {
+            stats: stats.clone(),
+            variant: if stats.lock().unwrap().is_link_up(rtt) {
+                StateVariant::Up
+            } else {
+                StateVariant::Down
+            },
         }
     }
 
-    fn update(
-        &mut self,
-        rx_bytes: Option<u64>,
-        tx_bytes: Option<u64>,
-        node_state: NodeState,
-        rtt: Duration,
-        ping_enabled: bool,
-    ) -> StateUpdateResult {
-        // Guard if node_state is not connected
-        // self -> new with down and return
-        if !matches!(node_state, NodeState::Connected) {
-            *self = State::new(rx_bytes, tx_bytes, node_state);
-            return Self::build_result(StateDecision::NoAction, LinkState::Down);
-        }
-
-        let new_rx = rx_bytes.unwrap_or_default();
-        let new_tx = tx_bytes.unwrap_or_default();
-        self.stats.update(new_rx, new_tx);
-
-        let is_link_up = self.stats.is_link_up(rtt);
+    fn update(&mut self, rtt: Duration, ping_enabled: bool) -> StateUpdateResult {
```

Comment:
```suggestion
    fn update(&mut self, cfg_max_allowed_rtt: Duration, ping_enabled: bool) -> StateUpdateResult {
```
same as https://github.com/NordSecurity/libtelio/pull/639/files#r1665336881

Reply:
same here